crates/storage-query-datafusion/src/partition_state/schema.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2023 - 2025 Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+#![allow(dead_code)]
+
+use datafusion::arrow::datatypes::DataType;
+
+use crate::table_macro::*;
+
+define_table!(
+    /// Observed partition state
+    partition_state(
+        /// Partition ID
+        partition_id: DataType::UInt32,
+
+        /// Node ID where the partition is running
+        plain_node_id: DataType::Utf8,
+
+        /// Node generation
+        gen_node_id: DataType::Utf8,
+
+        /// Observed target run mode of partition (LEADER, FOLLOWER)
+        target_mode: DataType::Utf8,
+
+        /// Effective partition run mode of partition (LEADER, FOLLOWER)
+        effective_mode: DataType::Utf8,
+
+        /// Last updated
+        updated_at: DataType::UInt64,
+
+        /// Last observed leader epoch
+        leader_epoch: DataType::UInt64,
+
+        /// Last observed leader node ID
+        leader_node_id: DataType::UInt32,
```

Comment:
Similar to sequencer, I highly doubt that the plain node id of the leader is useful, but I could be proven wrong if you have an example. I'd just replace it with `leader` or `leader_node_id` depends on how you call `sequencer`.